### DISCUSSION TITLE
Status in HA only error or running?

### DISCUSSION THREAD

**Discussion by simondsmason** [blakeblackshear/frigate#9032] [Ask A Question]
This might be a Home Assistant question so please let me know if I should post over there.

I have an automation that runs when Frigate is started to confirm camera settings - recording, events, etc.  I have one camera that is misbehaving and throwing errors so it causing frigate status to change from running to error and back again.  Separately I am troubleshooting this camera.  Since my trigger on startup is change to running it is causing this automation to run constantly.

So I thought I would be smart and change the trigger to not fire when changing from error to running.  Turns out that the only two states the frigate status are reporting is running, or error for anything else - including when frigate is shutdown.  In the dropdown of states I see unavailable and unknown - but it seems that these statuses are never reported.  And running and error don't appear in the picker.

So my goal of having an automation trigger except when changing from error to running doesn't appear possible?  Thanks.

**Comment by NickM-27** [comment]
a camera issue wouldn't change the server status, the only thing that would change the server status from running to error is if frigate was entirely unreachable because it stopped or restarted.

It is correct that those are the only two expected states

**Comment by simondsmason** [comment]
Thanks - so frigate server down - error.  Unavailable and unknown are red herrings.  And I should not be getting error during normal operation with server running - so let me explore what is happening there.

Also, I just want to say that your dedication to this project is unbelievable.  I moved off Blue Iris as it was crashing on me constantly and the UI is like a treasure hunt.  I really enjoyed setting up Frigate and still have many options to take advantage of.  I am building a new Unraid server and looking forward to configuring everything.  Thank you for all you do.

**Reply by NickM-27** [reply]
Correct, what exactly are you looking to use this status for? There may be a better option. 

And thanks, I really appreciate that.

**Comment by simondsmason** [comment]
I have/had an issue with cameras not retaining their switched states on restart of frigate.  Some cameras were for example recording on frigate restart and I set up an automation to confirm that they are turned off by default.  To be honest I have not actually confirmed that this is still happening so will look into that and whether this is necessary.

A second need - I send a refresh to my touchscreens in the house displaying camera feeds.  Many of them do not immediately refresh the camera feeds when frigate goes down.  Not at all mission critical - just a nice to have!

I will look into the error later today as now I know that the server should not be reporting an error state I want to get to the bottom of that.  Thanks.

**Reply by NickM-27** [reply]
1. This is true. I do this by having an automation that triggers when the frigate recording switch goes from unavailable to any other state.

**Reply by simondsmason** [reply]
I found the possible cause of the errors.  Running out of tmpfs space.  I increased this to 2GB - rather arbitrarily.  I see another post on this - https://github.com/blakeblackshear/frigate/issues/3127 - so will post on there for more direction on best approach to making sure I solve this correctly.

What recording switch is that you are referring to?  I can't find an entity that matches that.  Thanks.

**Reply by NickM-27** [reply]
by recording switch i mean switch.camera_name_record

**Reply by simondsmason** [reply]
I don't see any switch entity showing with switch. so not sure where to look for this in HA?  Thanks.

**Reply by NickM-27** [reply]
do you have the HA integration installed? it is right there
![Screen Shot 2023-12-21 at 09 11 17 AM](https://github.com/blakeblackshear/frigate/assets/14866235/7b1be1fa-dc11-412d-803f-768c856ecb45)

**Reply by simondsmason** [reply]
This might have something to do with it!

![image](https://github.com/blakeblackshear/frigate/assets/6063262/8e5940b6-7f74-488a-a40b-43b4ded447f1)

Will solve this and hopefully be off to the races.  I am still getting the underlying error that caused all of this conversation but posted on a related post - https://github.com/blakeblackshear/frigate/issues/5403
